am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java

```diff
@@ -241,8 +250,29 @@ public FilteredResourceEnvelope filterResource(@NotBlank String userId,
     @SuppressWarnings("LineLength")
     @AuditLog("returned resources that user with roles '{{userRoles}}' has create permission to: {{result}}")
     public Set<ResourceDefinition> getResourceDefinitionsWithRootCreatePermission(@NotEmpty Set<@NotBlank String> userRoles) {
-        return jdbi.withExtension(AccessManagementRepository.class, dao ->
-            dao.getResourceDefinitionsWithRootCreatePermission(userRoles));
+        Set<ResourceAttribute> resourceAttributes = jdbi.withExtension(AccessManagementRepository.class, dao ->
+            dao.getResourceAttributesWithRootCreatePermission(userRoles));
+
+        if (resourceAttributes.isEmpty()) {
+            return null;
+        }
+
+        Integer maxSecurityClassificationForRole = jdbi.withExtension(AccessManagementRepository.class, dao ->
+            dao.getRoles(userRoles, AccessType.ROLE_BASED)
+                .stream()
+                .mapToInt(role -> role.getSecurityClassification().getHierarchy())
+                .max().orElseThrow(NoSuchElementException::new));
+
+        Set<ResourceDefinition> filteredResourceDefinitions = resourceAttributes.stream()
+            .filter(resourceAttribute ->
+                resourceAttribute.getDefaultSecurityClassification().getHierarchy() <= maxSecurityClassificationForRole)
```

Comment:
Please extract method similar to `Permission#isGranted(int permissions)` that tests whether given security classification is visible for given highest security classification.

```
    public boolean isVisible(int maxHierarchy) {
        return (permissions & this.getValue()) == this.getValue();
    }
```

Having this I'd like to have unit tests that test every single combination. This way we won't need to test it in Integration tests.

Reply:
added parameterized unit test